src/lib/operators/join_hash.cpp

```diff
@@ -76,7 +77,18 @@ std::shared_ptr<const Table> JoinHash::_on_execute() {
   _impl = make_unique_by_data_types<AbstractReadOnlyOperatorImpl, JoinHashImpl>(
       build_input->column_data_type(build_column_id), probe_input->column_data_type(probe_column_id), build_operator,
       probe_operator, _mode, adjusted_column_ids, _predicate_condition, inputs_swapped);
-  return _impl->_on_execute();
+  auto output_table = _impl->_on_execute();
+
+  /**
+   * Defragment the output table - this is a performance optimization done because JoinHash is notorious for producing
+   * many Chunks (>500 seen) and performing badly on many Chunks (see loop nesting in _partition_radix_parallel)
+   */
+
+  // Somewhat arbitrarily chose min chunk size for the defragmentation. We want no Chunks smaller than that.
+  constexpr auto min_chunk_size = 10'000;
```

Comment:
How do you feel about a prefix for configuration parameters? @Bensk1 should like this because it makes it easier when he builds his self-driving tuning robot. Maybe `knob_`?